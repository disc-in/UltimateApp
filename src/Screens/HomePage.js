import React, { useLayoutEffect } from 'react';
import { StyleSheet, View, Text, ImageBackground, TouchableHighlight } from 'react-native';
import { createMaterialBottomTabNavigator } from '@react-navigation/material-bottom-tabs';
import { Ionicons, MaterialCommunityIcons } from '@expo/vector-icons';

import I18n from '../utils/i18n';
import theme from '../styles/theme.style';
import frisbeeGlove from '../../assets/HomePage/frisbeeglove.jpg';
import bodyweight from '../../assets/HomePage/bodyweight.jpg';
import gymstrong from '../../assets/HomePage/gymstrong.jpg';
import leanfit from '../../assets/HomePage/leanfit.jpg';
import simulator from '../../assets/HomePage/simulator.jpg';
import dictionary from '../../assets/HomePage/dictionary.jpg';
import ourPlays from '../../assets/HomePage/ourplays.jpg';
import essential from '../../assets/HomePage/essential.jpg';
import adultPrograms from '../../assets/HomePage/adultPrograms.png';
import juniorPrograms from '../../assets/HomePage/juniorPrograms.png';
import HeaderButton from '../Components/shared/HeaderButton';
import { AgeCategory, DrillTypes, EquipmentLabels } from '../Fixtures/config';

const Tab = createMaterialBottomTabNavigator();

const Frisbee = (props) => {
  return (
    <View style={styles.mainContainer}>
      <TouchableHighlight
        onPress={() => props.navigation.navigate('DrillListPage', { type: DrillTypes.FRISBEE })}
        style={styles.menuItem}
      >
        <ImageBackground source={frisbeeGlove} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.drills')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>

      <View style={styles.containerGrow}>
        <TouchableHighlight
          onPress={() =>
            props.navigation.navigate('ProgramListPage', { type: DrillTypes.FRISBEE, ageCategory: AgeCategory.JUNIOR })
          }
          style={styles.menuItemSplitLeft}
        >
          <ImageBackground source={juniorPrograms} style={styles.image}>
            <View style={styles.wrapper}>
              <Text style={styles.title}>{I18n.t('homePage.junior')}</Text>
              <Text style={styles.subtitle}>{I18n.t('homePage.programs')}</Text>
            </View>
          </ImageBackground>
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() =>
            props.navigation.navigate('ProgramListPage', { type: DrillTypes.FRISBEE, ageCategory: AgeCategory.SENIOR })
          }
          style={styles.menuItemSplitRight}
        >
          <ImageBackground source={adultPrograms} style={styles.image}>
            <View style={styles.wrapperRight}>
              <Text style={styles.titleRight}>{I18n.t('homePage.adult')}</Text>
              <Text style={styles.subtitleRight}>{I18n.t('homePage.programs')}</Text>
            </View>
          </ImageBackground>
        </TouchableHighlight>
      </View>

      <TouchableHighlight onPress={() => props.navigation.navigate('PlaybookPage')} style={styles.menuItem}>
        <ImageBackground source={ourPlays} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.playbook')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
    </View>
  );
};

const Fitness = (props) => {
  return (
    <View style={styles.mainContainer}>
      <TouchableHighlight
        onPress={() => props.navigation.navigate('DrillListPage', { type: DrillTypes.FITNESS })}
        style={styles.menuItem}
      >
        <ImageBackground source={leanfit} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.leanTitle')}</Text>
            <Text style={styles.subtitle}>{I18n.t('homePage.leanSubtitle')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
      <TouchableHighlight
        onPress={() =>
          props.navigation.navigate('ProgramListPage', {
            type: DrillTypes.FITNESS,
            equipmentLabel: EquipmentLabels.NONE,
          })
        }
        style={styles.menuItem}
      >
        <ImageBackground source={bodyweight} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.bodyweightTitle')}</Text>
            <Text style={styles.subtitle}>{I18n.t('homePage.bodyweightSubtitle')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
      <TouchableHighlight
        onPress={() =>
          props.navigation.navigate('ProgramListPage', {
            type: DrillTypes.FITNESS,
            equipmentLabel: EquipmentLabels.FULL,
          })
        }
        style={styles.menuItem}
      >
        <ImageBackground source={gymstrong} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.gymTitle')}</Text>
            <Text style={styles.subtitle}>{I18n.t('homePage.gymSubtitle')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
    </View>
  );
};

const Theory = (props) => {
  return (
    <View style={styles.mainContainer}>
      <TouchableHighlight onPress={() => props.navigation.navigate('DictionaryPage')} style={styles.menuItem}>
        <ImageBackground source={dictionary} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.dictionary')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
      <TouchableHighlight
        onPress={() => props.navigation.navigate('EssentialPage', { type: DrillTypes.FRISBEE })}
        style={styles.menuItem}
      >
        <ImageBackground source={essential} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.essential')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
      <TouchableHighlight onPress={() => props.navigation.navigate('TacticsPage')} style={styles.menuItem}>
        <ImageBackground source={simulator} style={styles.image}>
          <View style={styles.wrapper}>
            <Text style={styles.title}>{I18n.t('homePage.tactics')}</Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
    </View>
  );
};

export default HomePage = (props) => {
  const { navigation } = props;

  useLayoutEffect(() =>
    navigation.setOptions({
      headerRight: () => (
        <HeaderButton
          icon="information-outline"
          onPress={() => props.navigation.navigate('AboutPage')}
          testID="aboutButton"
        />
      ),
    }),
  );

  return (
    <Tab.Navigator initialRouteName="Frisbee">
      <Tab.Screen
        name="Frisbee"
        component={Frisbee}
        options={{
          tabBarLabel: I18n.t('homePage.frisbeeTab'),
          tabBarIcon: ({ color }) => <Ionicons name="ios-disc" color={color} size={22} />,
        }}
      />
      <Tab.Screen
        name="Fitness"
        component={Fitness}
        options={{
          tabBarLabel: I18n.t('homePage.fitnessTab'),
          tabBarIcon: ({ color }) => <Ionicons name="ios-fitness" color={color} size={22} />,
        }}
      />
      <Tab.Screen
        name="Theory"
        component={Theory}
        options={{
          tabBarLabel: I18n.t('homePage.theoryTab'),
          tabBarIcon: ({ color }) => <MaterialCommunityIcons name="bookshelf" color={color} size={22} />,
        }}
      />
    </Tab.Navigator>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    height: '100%',
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
  },
  imageContainer: {
    paddingTop: 20,
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
  },
  menuItem: {
    flexBasis: '25%',
    flexGrow: 1,
    marginBottom: 5,
  },
  menuItemSplitLeft: {
    flexBasis: '25%',
    flexGrow: 1,
    marginBottom: 5,
    marginRight: 5,
    width: '50%',
  },
  menuItemSplitRight: {
    flexBasis: '25%',
    flexGrow: 1,
    marginBottom: 5,
    width: '50%',
  },
  containerGrow: {
    flexBasis: '25%',
    flexGrow: 1,
    flex: 1,
    flexDirection: 'row',
  },
  image: {
    height: '100%',
  },
  title: {
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: 28,
    fontWeight: 'bold',
  },
  subtitle: {
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  wrapper: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 20,
  },
  wrapperRight: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center',
    alignItems: 'flex-end',
    padding: 20,
  },
  titleRight: {
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: 28,
    fontWeight: 'bold',
    textAlign: 'right',
  },
  subtitleRight: {
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: theme.FONT_SIZE_MEDIUM,
    textAlign: 'right',
  },
});
