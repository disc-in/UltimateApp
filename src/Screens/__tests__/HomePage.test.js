import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import HomePage from '../HomePage';
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('<HomePage />', () => {
  it('renders correctly', async () => {
    const navigation = { setOptions: jest.fn() };

    const { toJSON } = await waitFor(() =>
      render(
        <NavigationContainer>
          <HomePage navigation={navigation} />
        </NavigationContainer>,
      ),
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('links to about page', async () => {
    const navigate = jest.fn();

    const Stack = createStackNavigator();
    const { getByTestId } = await waitFor(() =>
      render(
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              name="HomePage"
              component={HomePage}
              listeners={({ navigation }) => ({
                transitionStart: (e) => {
                  navigation.navigate = navigate;
                },
              })}
            />
          </Stack.Navigator>
        </NavigationContainer>,
      ),
    );

    fireEvent.press(getByTestId('aboutButton'));

    expect(navigate).toBeCalledWith('AboutPage');
  });
});
