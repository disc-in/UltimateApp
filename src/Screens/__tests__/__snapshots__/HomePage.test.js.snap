// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        collapsable={false}
        importantForAccessibility="auto"
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "display": "flex",
            },
          ]
        }
        testID="RouteScreen: 0"
      >
        <View
          collapsable={false}
          style={
            Object {
              "flex": 1,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "height": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexBasis": "25%",
                  "flexGrow": 1,
                  "marginBottom": 5,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": undefined,
                      },
                      undefined,
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "padding": 20,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 28,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Drills
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexBasis": "25%",
                  "flexDirection": "row",
                  "flexGrow": 1,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexBasis": "25%",
                    "flexGrow": 1,
                    "marginBottom": 5,
                    "marginRight": 5,
                    "width": "50%",
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <Image
                    source={1}
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "height": "100%",
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "padding": 20,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 28,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Junior
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        }
                      }
                    >
                      Programs
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexBasis": "25%",
                    "flexGrow": 1,
                    "marginBottom": 5,
                    "width": "50%",
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <Image
                    source={1}
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "height": "100%",
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "padding": 20,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 28,
                          "fontWeight": "bold",
                          "textAlign": "right",
                        }
                      }
                    >
                      Senior
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                          "textAlign": "right",
                        }
                      }
                    >
                      Programs
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexBasis": "25%",
                  "flexGrow": 1,
                  "marginBottom": 5,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "height": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": undefined,
                      },
                      undefined,
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "padding": 20,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 28,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    My Playbook
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "#232b2b",
          "bottom": 0,
          "elevation": 4,
          "left": 0,
          "position": null,
          "right": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#232b2b",
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="tablist"
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "marginBottom": 0,
                "marginHorizontal": 0,
                "maxWidth": 504,
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#A6A6A6",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    collapsable={false}
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#fff",
                          },
                        ],
                      ]
                    }
                  >
                    Frisbee
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#A6A6A6",
                          },
                        ],
                      ]
                    }
                  >
                    Frisbee
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#A6A6A6",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    collapsable={false}
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#fff",
                          },
                        ],
                      ]
                    }
                  >
                    Fitness
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#A6A6A6",
                          },
                        ],
                      ]
                    }
                  >
                    Fitness
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "material-community",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󱉟
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#A6A6A6",
                          "fontSize": 22,
                        },
                        undefined,
                        Object {
                          "fontFamily": "material-community",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󱉟
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    collapsable={false}
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#fff",
                          },
                        ],
                      ]
                    }
                  >
                    Theory
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "fontSize": 12,
                            "height": 56,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "#A6A6A6",
                          },
                        ],
                      ]
                    }
                  >
                    Theory
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
